pageserver/src/tenant/secondary/heatmap.rs

```diff
@@ -15,6 +15,14 @@ pub(super) struct HeatMapTenant {
     pub(super) generation: Generation,
 
     pub(super) timelines: Vec<HeatMapTimeline>,
+
+    /// Uploaders provide their own upload period in the heatmap, as a hint to downloaders
+    /// of how frequently it is worthwhile to check for updates.
+    ///
+    /// This is optional for backward compat, and because we sometimes might upload
+    /// a heatmap explicitly via API for a tenant that has no periodic upload configured.
+    #[serde(default)]
+    pub(super) upload_period_ms: Option<u128>,
```

Comment:
Change to heatmap structure will change the content bytes and every tenant will have to reupload the new version. Will it cause too much load?

Reply:
We already do uploads of all heatmaps after a restart (jittered across the heatmap period). 
 `heatmap_upload_concurrency` (default 8) limits this to prevent overwhelming anything.